src/routes/safe/components/Apps/hooks/useIframeMessageHandler.ts

```diff
@@ -1,52 +1,131 @@
+  closeModal: () => void,
+  iframeRef: MutableRefObject<HTMLIFrameElement>,
+): ReturnType => {
+  const { enqueueSnackbar, closeSnackbar } = useSnackbar()
+  const safeName = useSelector(safeNameSelector)
+  const safeAddress = useSelector(safeParamAddressFromStateSelector)
+  const ethBalance = useSelector(safeEthBalanceSelector)
+  const network = useSelector(networkSelector)
+  const dispatch = useDispatch()
+
+  const sendMessageToIframe = useCallback(
+    function <T extends keyof InterfaceMessages>(messageId: T, data: InterfaceMessageToPayload[T]) {
+      if (iframeRef?.current && selectedApp) {
+        iframeRef.current.contentWindow.postMessage({ messageId, data }, selectedApp.url)
+      }
+    },
+    [iframeRef, selectedApp],
+  )
 
-const useIframeMessageHandler = (): void => {
   useEffect(() => {
-    //   const handleIframeMessage = (data) => {
-    //     if (!data || !data.messageId) {
-    //       console.error('ThirdPartyApp: A message was received without message id.')
-    //       return
-    //     }
-    //     switch (data.messageId) {
-    //       case operations.SEND_TRANSACTIONS: {
-    //         const onConfirm = async () => {
-    //           closeModal()
-    //           await sendTransactions(dispatch, safeAddress, data.data, enqueueSnackbar, closeSnackbar, selectedApp.id)
-    //         }
-    //         confirmTransactions(
-    //           safeAddress,
-    //           safeName,
-    //           ethBalance,
-    //           selectedApp.name,
-    //           selectedApp.iconUrl,
-    //           data.data,
-    //           openModal,
-    //           closeModal,
-    //           onConfirm,
-    //         )
-    //         break
-    //       }
-    //       default: {
-    //         console.error(`ThirdPartyApp: A message was received with an unknown message id ${data.messageId}.`)
-    //         break
-    //       }
-    //     }
-    //   }
-    //   const onIframeMessage = async ({ data, origin }) => {
-    //     if (origin === window.origin) {
-    //       return
-    //     }
-    //     if (!selectedApp.url.includes(origin)) {
-    //       console.error(`ThirdPartyApp: A message was received from an unknown origin ${origin}`)
-    //       return
-    //     }
-    //     handleIframeMessage(data)
-    //   }
-    //   window.addEventListener('message', onIframeMessage)
-    //   return () => {
-    //     window.removeEventListener('message', onIframeMessage)
-    //   }
-  }, [])
+    const handleIframeMessage = (msg: CustomMessageEvent) => {
+      if (!msg?.data.messageId) {
+        console.error('ThirdPartyApp: A message was received without message id.')
+        return
+      }
+      switch (msg.data.messageId) {
+        case SDK_MESSAGES.SEND_TRANSACTIONS: {
+          const onConfirm = async () => {
+            closeModal()
+            await sendTransactions(dispatch, safeAddress, msg.data.data, enqueueSnackbar, closeSnackbar, selectedApp.id)
```

Comment:
Calling this function here makes the effect to receive a lot of dependencies, could it be possible to add a new function that receives as argument the values related to the message and then calls to sendTransaction?

Reply:
yes I agree that doesn't seem nice, but I couldn't come up with a good solution

> add a new function that receives as argument the values related to the message and then calls to sendTransaction

but it still means that all the values have to be a dependency somewhere, isn't it? If I got you wrong, could you please give an example?

Reply:
didn't test it but I was thinking about something like this:
```
...
  closeModal()
  processSendTransactionMessage(msg.data.data)
  break
...
```

and then outside the effect, in processSendTransaction make the call to sendTransaction. Perhaps the use of useCallback could be needed.